File: Logic_HandsOn/C++/Problems on Digits,String,Array,Pointers etc/program28.cpp
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		a program which accept range from user and return addition of all numbers in between that range. 
//		(Range should contains positive numbers only)
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////

#include<iostream>
using namespace std;

class Range
{
	private:
		int iStart;
		int iEnd;
		int iCnt;
		int iSum;

	public:
		Range(int iStart=0,int iEnd=0,int iCnt=0,int iSum=0)
		{
		
		}

		int RangeSum()
		{
        		if (iStart>iEnd || iStart<0)
        		{
                		cout<<"Invalid Input\n";
        		}
        		else
        		{
                		for(iCnt=iStart;iCnt<=iEnd;iCnt++)
                		{
                      			iSum = iSum+iCnt;
                		}
        		}

			return iSum;
		}
};		

int main()
{
        int iValue1=0,iValue2=0,iRet=0;

        cout<<"Enter starting point ";
	cin>>iValue1;
        
	cout<<"Enter ending point ";
	cin>>iValue2;
	
        Range *rObj = new Range(iValue1,iValue2);
        iRet = rObj->RangeSum();

	if(iRet!=0)
	{
		cout<<"Addition is "<<iRet<<"\n";
	}	

        return 0;
}

File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program28.java
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		a program which accept range from user and return addition of all numbers in between that range. 
//		(Range should contains positive numbers only)
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.*;

class Solution
{
	public int iStart,iEnd;

	public Solution(int iStart,int iEnd)
	{
		this.iStart = iStart;
		this.iEnd = iEnd;
	}

	public int RangeSum()
	{

  		int iSum=0;		
		if(iStart<0 || iEnd<0)
		{
			System.out.println("Invalid Syntax");
			System.exit(0);
		}		

		for(int iCnt=iStart;iCnt<=iEnd;iCnt++)
		{
			iSum=iSum+iCnt;
		}

		return iSum;
	
	}
}

class Program
{
	public static void main(String[] args)
	{
		Scanner sObj = new Scanner(System.in);

		int iRet=0;
		System.out.println("Enter Start: ");
		int iStart = sObj.nextInt();
		
		System.out.println("Enter End: ");
		int iEnd = sObj.nextInt();

		Solution obj = new Solution(iStart,iEnd);
		iRet = obj.RangeSum();
	
		System.out.println("Addition is: "+iRet);
	}
}

/*
OUTPUT:

Enter Start: 
23
Enter End: 
30
Addition is: 212

*/
